### ISSUE TITLE
Guided Tour: Step visibility and positioning bug in Tabbed sections

### ISSUE THREAD

**Issue by donpark** [Automattic/wp-calypso#33366]
How to reproduce the bug:

1. Reduce browser tab window height to something small, like 800.
2. Navigate to an Atomic or Jetpack site's URL https://wordpress.com/settings/security/{site_slug}?tour=jetpackMonitoring. You should see a Tour Step under "Monitor your site's downtime" checkbox.

<img width="480" alt="screenshot_683" src="https://user-images.githubusercontent.com/127594/58447514-7282f180-80b9-11e9-91ff-dc95e4a71d4f.png">

3. Click on Performance tab, then scroll down. You'll see the Step from the other tab displayed when it shouldn't.

<img width="867" alt="screenshot_682" src="https://user-images.githubusercontent.com/127594/58447494-61d27b80-80b9-11e9-8bbd-fc68cdf55cb2.png">

**Comment by simison** [comment]
Cc @automattic/luna because this is used in the checklist

**Comment by simison** [comment]
Would using `onTargetDisappear` prop help here?

https://github.com/Automattic/wp-calypso/blob/59bdfeeb97eda4266ad39410cb0a074d2c88dbc8/client/layout/guided-tours/docs/API.md#props-1

**Comment by donpark** [comment]
> Would using onTargetDisappear prop help here?

Yes if old targets are not rendered when tab switches. That's the norm for React but jQuery norm was to just hide or move offscreen.

**Comment by donpark** [comment]
Well, I just tried it and it didn't quite work as expected.

1. onTargetDisappear handler that quits if target disappears will quit the tour when tab is changed but step did not re-appear without refresh.
2. Tab URLs do not preserve the active tour so switching tab puts URL in a confused state.

**Comment by davipontesblog** [comment]
Moved to Jetpack Improved Onboarding for triaging.
